core/src/test/scala/unit/kafka/log/LogCleanerTest.scala

```diff
@@ -137,10 +137,38 @@ class CleanerTest extends JUnitSuite {
                (0 until leo.toInt by 2).forall(!keys.contains(_)))
   }
 
+  def testLogCleanerStats(): Unit = {
```

Comment:
Not really, this was mainly for completeness (we didn't seem to have much testing around cleaner stats). I guess we can add another test which asserts that `lastStats` is not affected by a call to `clean()`, but it seems like overkill given the refactor. What do you think?